server/pubsub/redis_query_results.go

```diff
@@ -90,9 +90,16 @@ func receiveMessages(ctx context.Context, conn *redigo.PubSubConn, outChan chan<
 	defer close(outChan)
 
 	for {
-		// Add a timeout to try to cleanup in the case the server has somehow gone completely unresponsive.
+		beforeReceive := time.Now()
+
+		// Add a timeout to try to cleanup in the case the server has somehow gone
+		// completely unresponsive.
 		msg := conn.ReceiveWithTimeout(1 * time.Hour)
 
+		if recvTime := time.Since(beforeReceive); recvTime > time.Minute {
```

Comment:
Even a few seconds is probably long and unusual, but I went a bit on the safer side. In all likeliness, even if we see this log, unless we see it with a lot of different connections (indicating that many connections are blocked/wasted on this), it shouldn't be the source of problems.